libraries/standard/mqtt/src/mqtt.c

```diff
@@ -396,7 +401,8 @@ static bool matchEndWildcardsSpecialCases( const char * pTopicFilter,
      * "/#". This check handles the case to match filter "sport/#" with topic
      * "sport". The reason is that the '#' wildcard represents the parent and
      * any number of child levels in the topic name.*/
-    if( ( filterIndex == ( topicFilterLength - 3U ) ) &&
+    if( ( topicFilterLength >= 3U ) &&
```

Comment:
Added `( topicFilterLength >= 3U )` condition to address overflow error flagged by CBMC proof for `( filterIndex == ( topicFilterLength - 3U ) )` condition